src/dsp_tools/utils/connection_live.py

```diff
@@ -395,5 +310,57 @@ def delete(
             headers=headers,
             timeout=timeout,
         )
-        check_for_api_error(response)
         return cast(dict[str, Any], response.json())
+
+    def _try_network_action(self, action: Callable[..., Any]) -> Any:
```

Comment:
`self` is not called anywhere in this method, so this could just as well stay a top level function, I think

Reply:
ok, that makes sense

Reply:
in theory you could pass the session object to the function as an argument. But I'm absolutely ok with making it a method, I just didn't see the reason